compiler-core/src/language_server/configuration.rs

```diff
@@ -0,0 +1,45 @@
+use serde::Deserialize;
+use std::sync::{Arc, RwLock};
+
+pub type SharedConfig = Arc<RwLock<Configuration>>;
+
+#[derive(Debug, Default, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Configuration {
+    pub inlay_hints: InlayHintsConfig,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InlayHintsConfig {
+    #[serde(default = "InlayHintsConfig::default_function_definitions")]
+    pub function_definitions: bool,
+
+    #[serde(default = "InlayHintsConfig::default_module_constants")]
+    pub module_constants: bool,
+
+    #[serde(default = "InlayHintsConfig::default_variable_assignments")]
+    pub variable_assignments: bool,
+}
```

Comment:
Could you document in detail what these do please 🙏

Reply:
Yep, will do